posthog/batch_exports/service.py

```diff
@@ -932,3 +932,21 @@ def get_is_backfill(self) -> bool:
         if self.backfill_details is not None:
             return True
         return self.is_backfill
+
+    def properties_to_log(self) -> list[str]:
+        """Return a dictionary of properties that we want to log if an error is raised.
+
+        We list these explicitly rather than setting all fields as safe to log, so that the default is opt-out (just in
+        case new fields get added which are sensitive).
+        """
```

Comment:
Could this just be an attribute or a property?

Reply:
yeah, I could make it a property. I chose using a method over an attribute so it was a bit more flexible and we could use inheritance (eg for some of the batch export classes)